Fix selection of date-time in Final Form

jamesricky opened the pull request:
## Description

Previously `onBlur` was passed from `Field` (from `react-final-form`) into the `Root` slot ( a `div`) of `DateTimePicker`. 
This caused unexpected behavior, preventing the selection of a date with a different month or year than the current value.

## Acceptance criteria

-   [x] I have verified if my change requires [an example](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md#example)
-   [x] I have verified if my change requires [a changeset](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md#changeset)
-   [x] I have verified if my change requires [screenshots/screencasts](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md#screenshotsscreencasts)

## Screenshots/screencasts

| Before | After |
| ------ | ----- |
| <video src="https://github.com/user-attachments/assets/34d6bc36-7107-4e24-a62c-2c4a1e9058ee" />   | <video src="https://github.com/user-attachments/assets/d956210e-c979-40dc-98dd-3cc7b10563c9" />  |

## Open TODOs/questions

-   [x] Add changeset

## Further information

-   Task: https://vivid-planet.atlassian.net/browse/SVK-646

johnnyomair reviewed (commented):
Final Form uses [onFocus to mark the field as active](https://final-form.org/docs/react-final-form/types/FieldRenderProps#inputonfocus) and [onBlur to mark it as inactive](https://final-form.org/docs/react-final-form/types/FieldRenderProps#inputonblur). Would this still work? Should it be further down in the date-time picker?

jamesricky commented:
> Final Form uses [onFocus to mark the field as active](https://final-form.org/docs/react-final-form/types/FieldRenderProps#inputonfocus) and [onBlur to mark it as inactive](https://final-form.org/docs/react-final-form/types/FieldRenderProps#inputonblur). Would this still work? Should it be further down in the date-time picker?

You're right. I reverted that change and added support for the `onBlur`/`onFocus` events to `DateTimePicker`. Those are now called when the event is triggered on either of the two sub-fields (DatePicker and TimePicker).
